scripts/python-skywater-pdk/skywater_pdk/fet_simulator/fet_simulator.py

```diff
@@ -0,0 +1,209 @@
+    figs[0].suptitle(f'{fet_name} Id/W vs gm/Id (W = {W})')
+    plts[0].set_xlabel("gm/Id")
+    plts[0].set_ylabel("Id/W")
+    plts[0].grid(True)
+    figs[1].suptitle(f'{fet_name} fT vs gm/Id (W = {W})')
+    plts[1].set_xlabel("gm/Id")
+    plts[1].set_ylabel("f_T")
+    plts[1].grid(True)
+    figs[2].suptitle(f'{fet_name} gm/gds vs gm/Id (W = {W})')
+    plts[2].set_xlabel("gm/Id")
+    plts[2].set_ylabel("gm/gds")
+    plts[2].grid(True)
+    figs[3].suptitle(f'{fet_name} gm/Id vs Vgg (W = {W})')
+    plts[3].set_xlabel("Vgg")
+    plts[3].set_ylabel("gm/Id")
+    plts[3].grid(True)
+    return figs, plts
+
+
+def gen_plots(gm_id, id_W, ft, gm_gds, vsweep, fet_W, fet_L, plts):
+    # plot some interesting things
+    plts[0].plot(gm_id, id_W, label=f'W {fet_W} x L {fet_L}')
+    plts[1].plot(gm_id, ft, label=f'W {fet_W} x L {fet_L}')
+    plts[2].plot(gm_id, gm_gds, label=f'W {fet_W} x L {fet_L}')
+    plts[3].plot(vsweep, gm_id, label=f'W {fet_W} x L {fet_L}')
+
+
+def read_bins(fname):
+    with open(fname, 'r') as f:
+        r = csv.reader(f)
+        # drop CSV header
+        next(r)
+        res = []
+        for line in r:
+            res.append(line)
+        return res
+
+
+def generate_fet_plots(
+        fet_type,
+        corner_path,
+        bins_csv,
+        outdir,
+        outprefix,
+        only_W=None,
+        ext='svg'):
+    print(f'[generate_fet_plots] {fet_type} {corner_path} {bins_csv}' +
+          f'{outdir} {outprefix} {only_W}')
+    iparam = f'@m.xm1.m{fet_type}[%s]'
+    # fet_W and fet_L values here are only for initialization, they are
+    # later changed in the for loop
+    c = create_test_circuit(fet_type, iparam, 0.15, 1, corner_path)
+    
+    bins = read_bins(bins_csv)
+
+    bins_by_W = defaultdict(list)
+    # group bins by W
+    for line in bins:
+        bins_by_W[line[2]].append(line)
+    
+    Ws = only_W if only_W is not None else list(bins_by_W.keys())
+
+    for W in Ws:
+        figs, plts = init_plots(fet_type, W)
+        for dev, bin, fet_W, fet_L in bins_by_W[W]:
+            fet_W, fet_L = float(fet_W), float(fet_L)
+            if only_W is not None and fet_W not in only_W:
+                continue
+            c.element('XM1').parameters['W'] = fet_W
+            c.element('XM1').parameters['L'] = fet_L
+            gm_id, ft, id_W, gm_gds, vsweep = run_sim(c, iparam, fet_W)
+            gen_plots(gm_id, id_W, ft, gm_gds, vsweep, fet_W, fet_L, plts)
+
+        figtitles = ['Id_w', 'fT', 'gm_gds', 'gm_id']
+        for fg, name in zip(figs, figtitles):
+            lg = fg.legend(
+                bbox_to_anchor=(1.05, 1),
+                loc='upper left'
+            )
+            fg.tight_layout()
+            fg.savefig(
+                Path(outdir) / (outprefix + f'_{name}_W{str(W).replace(".", "_")}.{ext}'),
+                bbox_extra_artists=(lg,),
+                bbox_inches='tight'
+            )
+        plt.close('all')
+
+
+if __name__ == '__main__':
```

Comment:
This should be in a `main()` function.

This bit should just be;
```python
if __name__ == '__main__':
    sys.exit(main(sys.argv))
```

Reply:
fixed